build/source/netcdf/def_output.f90

```diff
@@ -59,71 +60,84 @@ module def_output_module
  subroutine def_output(nHRU,nSoil,infile,err,message)
  USE globalData,only:structInfo                               ! information on the data structures
- USE globalData,only:forc_meta,attr_meta,type_meta            ! metadata structures
- USE globalData,only:prog_meta,diag_meta,flux_meta,deriv_meta ! metadata structures
- USE globalData,only:mpar_meta,indx_meta                      ! metadata structures
- USE globalData,only:bpar_meta,bvar_meta                      ! metadata structures
+ USE globalData,only:forc_meta,attr_meta,type_meta            ! metaData structures
+ USE globalData,only:prog_meta,diag_meta,flux_meta,deriv_meta ! metaData structures
+ USE globalData,only:mpar_meta,indx_meta                      ! metaData structures
+ USE globalData,only:bpar_meta,bvar_meta,time_meta            ! metaData structures
  USE globalData,only:model_decisions
+ USE globalData,only:ncid
+ USE globalData,only:nFreq,outFreq                            ! output frequencies
  USE multiconst,only:integerMissing
  ! declare dummy variables
- integer(i4b), intent(in)    :: nHRU                          ! number of HRUs
- integer(i4b), intent(in)    :: nSoil                         ! number of soil layers in the first HRU (used to define fixed length dimensions)
- character(*), intent(in)    :: infile                        ! file suffix
+ integer(i4b),intent(in)     :: nHRU                          ! number of HRUs
+ integer(i4b),intent(in)     :: nSoil                         ! number of soil layers in the first HRU (used to define fixed length dimensions)
+ character(*),intent(in)     :: infile                        ! file suffix
  integer(i4b),intent(out)    :: err                           ! error code
  character(*),intent(out)    :: message                       ! error message
  ! local variables
- integer(i4b)                :: ivar                          ! loop through model variables
- integer(i4b)                :: iStruct                       ! index of data structure
+ integer(i4b)                :: iVar                          ! loop through model variables
+ integer(i4b)                :: iFreq                         ! loop through output frequencies
+ integer(i4b)                :: iStruct                       ! loop through structure types 
+ integer(i4b),parameter      :: modelTime=1                   ! model timestep output frequency
+ character(len=5)            :: fstring                       ! string to hold model output freuqnecy
+ character(len=1000)         :: fname                         ! temporary filename
  character(len=256)          :: cmessage                      ! temporary error message
+
  ! initialize errors
  err=0; message="def_output/"
 
  ! create initial file
- call ini_create(nHRU,nSoil,trim(infile),err,cmessage)
- if(err/=0)then; message=trim(message)//trim(cmessage); return; endif
+ do iFreq = 1,nFreq
+  write(fstring,'(i5)') outFreq(iFreq)
+  fstring = adjustl(fstring)
```

Comment:
Yes, it could be done. notice that it will require a decimal point in the file name. There are two ways that we could organize the output timing specifications that *might* be more intuitive to users.

1. To allow the use to specify the output frequency in terms of months, weeks, days, hours, minutes, seconds.

2. To allow the user to specify specific output times during each month, day, hour, minute. 

LIS uses the first option, and this works well unless you want output at a different time than midnight (say, for instantaneous outputs). So I really prefer not to go this route, because it is quite limiting. The second route is better, but requires quite detailed lists of specifications for output streams. Notice that the timestep option is incredibly simple - both in terms of code and in terms of intuition. I actually prefer this way as a user, even if I have to do a little math on the user side. Anyway, let's have a short chat about this before we change anything.

One thing I don't think we would want to do is to have the output file labeled in a way that is different than how the user is asked to specify output frequency.